integration/src/main/java/scripts/testscripts/WorkspacePermissionOperations.java

```diff
@@ -0,0 +1,173 @@
+  }
+  // TODO (DC-446): Fix with https://broadworkbench.atlassian.net/browse/DC-446
+  //  private void testNoAccessPermissions() throws Exception {
+  //    setTestWorkspacePermissionForRegularUser(WorkspaceAccessLevel.NO_ACCESS.getValue());
+  //    // User cannot create a catalog entry on Workspace when user has "No Access" on a workspace
+  //    CreateDatasetRequest request =
+  // datasetRequestForWorkspace(adminTestWorkspace.getWorkspaceId());
+  //    // note if this assertion fails we'll leave behind a stray dataset in the db
+  //    assertThrows(ApiException.class, () -> userDatasetsApi.createDataset(request));
+  //    assertThat(
+  //        userDatasetsApi.getApiClient().getStatusCode(),
+  // is(HttpStatusCodes.STATUS_CODE_NOT_FOUND));
+  //    // but the user can get datasets
+  //    userDatasetsApi.getDataset(adminTestDatasetId);
+  //    assertThat(userDatasetsApi.getApiClient().getStatusCode(),
+  // is(HttpStatusCodes.STATUS_CODE_OK));
+  //  }
+
+  private void testAdminPermissionsOnUserSnapshot() throws Exception {
+    // Verify admin can create a dataset on the user snapshot
+    CreateDatasetRequest request = datasetRequestForWorkspace(userTestWorkspace.getWorkspaceId());
+    adminCreateDataset(request);
+  }
+
+  private void setTestWorkspacePermissionForRegularUser(String policy) throws Exception {
+    // first clear the shared snapshot of policy state caused by previous tests
+    adminRawlsClient.updateWorkspaceAcl(
+        List.of(new WorkspaceACLUpdate().email(regularUser.userEmail).accessLevel(policy)),
+        adminTestWorkspace);
+  }
+
+  private void clearTestWorkspacePermissions() throws Exception {
+    log.info(
+        "Clearing acl permissions for {}/{}",
+        adminTestWorkspace.getNamespace(),
+        adminTestWorkspace.getName());
+    setTestWorkspacePermissionForRegularUser(WorkspaceAccessLevel.NO_ACCESS.getValue());
+  }
+
+  private CreateDatasetRequest datasetRequestForWorkspace(String workspaceId) {
+    final String metadata = """
+        {"name": "test"}""";
+    return new CreateDatasetRequest()
+        .catalogEntry(metadata)
+        .storageSourceId(workspaceId)
+        .storageSystem(StorageSystem.WKS);
+  }
+
+  private UUID adminCreateDataset(CreateDatasetRequest request) throws Exception {
+    var datasetId = adminDatasetsApi.createDataset(request).getId();
+    assertThat(adminDatasetsApi.getApiClient().getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+    datasetIds.add(datasetId);
+    return datasetId;
+  }
+
+  @Override
+  public void cleanup(List<TestUserSpecification> testUsers) throws Exception {
+    for (var datasetId : datasetIds) {
+      adminDatasetsApi.deleteDataset(datasetId);
+      log.info("deleted dataset {}", datasetId);
+    }
+    if (adminTestWorkspace != null) {
+      log.info(
+          "deleting workspace {}/{}",
+          adminTestWorkspace.getNamespace(),
+          adminTestWorkspace.getName());
```

Comment:
I think this information is already logged in `RawslClient.deleteWorkspace()`, and doesn't need to be logged here too.